tfx/components/example_gen/base_example_gen_executor_test.py

```diff
@@ -142,10 +143,97 @@ def testDoOutputSplit(self):
     # Check example gen outputs.
     self.assertTrue(tf.io.gfile.exists(self._train_output_file))
     self.assertTrue(tf.io.gfile.exists(self._eval_output_file))
+    
     # Output split ratio: train:eval=2:1.
     self.assertGreater(
         tf.io.gfile.GFile(self._train_output_file).size(),
         tf.io.gfile.GFile(self._eval_output_file).size())
+    
+  def testFeatureBasedPartition(self):
+    # Create exec proterties.
+    exec_properties = {
+        'input_config':
+            json_format.MessageToJson(
+                example_gen_pb2.Input(splits=[
+                    example_gen_pb2.Input.Split(
+                        name='single', pattern='single/*'),
+                ]),
+                preserving_proto_field_name=True),
+        'output_config':
+            json_format.MessageToJson(
+                example_gen_pb2.Output(
+                    split_config=example_gen_pb2.SplitConfig(splits=[
+                        example_gen_pb2.SplitConfig.Split(
+                            name='train', hash_buckets=2),
+                        example_gen_pb2.SplitConfig.Split(
+                            name='eval', hash_buckets=1)
+                    ], partition_feature_name='i')))
+    }
+
+    # Run executor.
+    example_gen = TestExampleGenExecutor()
+    example_gen.Do({}, self._output_dict, exec_properties)
+
+    # Check example gen outputs.
+    self.assertTrue(tf.io.gfile.exists(self._train_output_file))
+    self.assertTrue(tf.io.gfile.exists(self._eval_output_file))
+    
+    # Output split ratio: train:eval=2:1.
+    self.assertGreater(
+        tf.io.gfile.GFile(self._train_output_file).size(),
+        tf.io.gfile.GFile(self._eval_output_file).size())
+    
+  def testInvalidFeatureName(self):
+    # Create exec proterties.
+    exec_properties = {
+        'input_config':
+            json_format.MessageToJson(
+                example_gen_pb2.Input(splits=[
+                    example_gen_pb2.Input.Split(
+                        name='single', pattern='single/*'),
+                ]),
+                preserving_proto_field_name=True),
+        'output_config':
+            json_format.MessageToJson(
+                example_gen_pb2.Output(
+                    split_config=example_gen_pb2.SplitConfig(splits=[
+                        example_gen_pb2.SplitConfig.Split(
+                            name='train', hash_buckets=2),
+                        example_gen_pb2.SplitConfig.Split(
+                            name='eval', hash_buckets=1)
+                    ], partition_feature_name='invalid')))
+    }
+
+    # Run executor.
+    example_gen = TestExampleGenExecutor()
+    self.assertRaises(RuntimeError, example_gen.Do, 
```

Comment:
nit, to test exception, you can use 

```
with self.assertRaisesRegex(error_code, 'error message'):
   // code that triggers the error. 
```

with regex error message string, you can make sure the error is thrown at a particular callsite

using `with`, the you can include failure setup code together in the same block.